sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigurationReplicaClientsBuilder.java

```diff
@@ -175,10 +160,7 @@ List<AppConfigurationReplicaClient> buildClients(ConfigStore configStore) {
             for (String endpoint : endpoints) {
                 ConfigurationClientBuilder builder = this.createBuilderInstance();
                 if (!credentialConfigured) {
-                    // System Assigned Identity. Needs to be checked last as all of the above should
-                    // have an Endpoint.
-                    LOGGER.debug("Connecting to {} using Azure System Assigned Identity.", endpoint);
-                    builder.credential(new ManagedIdentityCredentialBuilder().build());
+                    builder.credential(new DefaultAzureCredentialBuilder().build());
```

Comment:
nit: do you think we should share this default DAC credential for all endpoints?

Reply:
Good point. Updated.